editor.planx.uk/src/@planx/components/shared/Preview/NavigateToPublishedButton.tsx

```diff
@@ -0,0 +1,36 @@
+import Link from "@mui/material/Link";
+import Typography from "@mui/material/Typography";
+import React from "react";
+import { useNavigation } from "react-navi";
+
+import { InnerContainer } from "./SaveResumeButton";
```

Comment:
If this is now shared, it should be its own component. 

Generally we'd try to avoid feature to feature imports like this. The main reason being that when I make a change in future on this component it's not inherently obvious that it's going to have a knock on effect in an unrelated component.

A few options here - we could repeat the style, extract it to its own very simple component, or make a component which is the "or" text option which then takes in a callback function. I think the later is the best option here as it keeps an "or" text options consistent and we could control within the card that it only shows once and doesn't stack as is currently possible.

How about a component which took in a condition (to decide if it's shown or hidden) and a callback (to be called when clicked), which is then passed into the `Card`?

Reply:
Great comment, thanks for the suggestions here. @DafyddLlyr 

This may be an indulgence at the wrong time, but I've been reading a bit on decoupling code for maintainability and thus, taken a slightly different approach using the comment as inspo. 

I wanted to keep the handleClick logic within each `SaveAndResumeButton` and `NavigateToPublishedButton` so they could expand or become more complex in future without bloating out the `Card` and it means the components can focus on doing one thing well at a time. 

I have put in an extra component, `OrNavigationButton` to handle the logic of the displaying either button based on a `type` it receives from the `Card` component, so it again just does one wee thing well. The `Card` comp can then do a small function to define what type to show. 

Lemme know if I've led myself astray and you're suggestion is a cleaner way of doing what I was hoping to do!